stylefunction.js

```diff
@@ -66,7 +67,9 @@ export function getValue(layer, layoutOrPaint, property, zoom, feature) {
     let value = (layer[layoutOrPaint] || emptyObj)[property];
     const propertySpec = spec[`${layoutOrPaint}_${layer.type}`][property];
     if (value === undefined) {
-      value = propertySpec.default;
+      value = (propertySpec.default !== undefined) ? propertySpec.default : 'none';
```

Comment:
Leaving the value `undefined` here should not break anything, because a literal value will never run through a function conversion or expression evaluation.